src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -0,0 +1,367 @@
+                                                                         isReportedAtCompilationEnd: true);
+
+        internal static DiagnosticDescriptor CA2255Rule = DiagnosticDescriptorHelper.Create(CA2255RuleId,
+                                                                         s_localizableTitleCA2255,
+                                                                         s_localizableMessageCA2255,
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.BuildWarning,
+                                                                         description: s_localizableDescriptionCA2255,
+                                                                         isPortedFxCopRule: true,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(CA1727Rule, CA1848Rule, CA2253Rule, CA2254Rule, CA2255Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+
+                if (!wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerExtensions, out var loggerExtensionsType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingILogger, out var loggerType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerMessage, out var loggerMessageType))
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context => AnalyzeInvocation(context, loggerType, loggerExtensionsType, loggerMessageType), OperationKind.Invocation);
+            });
+        }
+
+        private void AnalyzeInvocation(OperationAnalysisContext context, INamedTypeSymbol loggerType, INamedTypeSymbol loggerExtensionsType, INamedTypeSymbol loggerMessageType)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+
+            var methodSymbol = invocation.TargetMethod;
+
+            if (methodSymbol.ContainingType == loggerExtensionsType)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(CA1848Rule, invocation.Syntax.GetLocation(), methodSymbol.Name));
+            }
```

Comment:
This reports a diagnostic on every usage of any method in this type. Why not mark this type obsolete in dotnet/runtime instead?

Reply:
you're suggesting to remove `CA1848` altogether? I'm not opposed to that

Reply:
Yes that's what I'm suggesting (but this requires obsoleting so a warning is still issued).

Reply:
@pakrym I chatted with @shirhatti as well and he mentioned he would be fine with skipping diagnostic `CA1848`. Since you were the original author, I wanted to make sure you are not against this either.

Reply:
Sounds good to me.

Reply:
Maybe we can turn it into a code-fix one day.